### DISCUSSION TITLE
DSM forward programming

### DISCUSSION THREAD

**Discussion by jken84** [pascallanger/DIY-Multiprotocol-TX-Module#871] [Q&A]
Can someone tell me where I can find the downloadable content for DSM forward programing. I see a reference to DSM+FwdPrg_05_BW.lua. But can not find any files to download. Thanks

**Comment by pascallanger** [comment]
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/tree/master/Lua_scripts#dsm-forward-programming

**Reply by jken84** [reply]
I have found the above link telling me what to copy. However I have not found the referenced material to download.

**Reply by pascallanger** [reply]
It's just above, all the files are at the top

**Reply by jken84** [reply]
Apparently I am missing something.  I see the files and folders.  However, there is no option to download the script. I can see the script but I don't see a downloadable file.  Thanks.

**Reply by jken84** [reply]
I may have figured it out.  You have to open each one and download them individually?

**Reply by pascallanger** [reply]
Download the full master zip file from the main page and extract what you need.

**Reply by jken84** [reply]
I'm sorry.  I do not see a downloadable zip file anywhere on the page.  All I see is a list of all the files.  I can open them up and see the content but no zip files to download.  Maybe there is a different page somewhere.  I must be over looking something.

**Reply by benlye** [reply]
This page:
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module

This button:
<img width="1076" alt="image" src="https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/23322579/445401a9-3d76-4e7f-9d5b-6352b5596abc">

After you click on the green "Code" button you can click on "Download Zip":
<img width="315" alt="image" src="https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/23322579/9b9fe597-e179-4cf2-8361-f35f0af88f8e">

**Reply by jken84** [reply]
Thank you….Got it.
